### PR TITLE
enh: Delete Application and its shares

### LINKED ISSUE #991: Applications: Allow users to delete applications
Leftover from #311 

We should offer users an option to delete applications

### PR DESCRIPTION
**Pull request by enjeck**
Fixes #991

### CODE REVIEW

**Comment by enjeck** [comment]
Need to check/fix some edge cases:
- [ ] Delete shares from a newly-created context without reload
- [ ] Delete shares from a context whose shares were updated
- [ ] check that transfer context works
- [ ]  check that shared resources are removed when context is deleted

File: src/modules/modals/EditContext.vue [hunk 1]
```diff
@@ -209,6 +223,37 @@ export default {
 			}
 			return resources
 		},
+		async actionDeleteContext() {
+			this.prepareDeleteContext = false
+			const res = await this.$store.dispatch('removeContext', { context: this.context, receivers: this.context.sharing })
```
Comment (juliushaertl) [inline src/modules/modals/EditContext.vue:228]: This throws an error as this.context doesn't seem to be defined in here. The following seems to work:

```
--- a/src/modules/modals/EditContext.vue
+++ b/src/modules/modals/EditContext.vue
@@ -225,7 +225,8 @@ export default {
                },
                async actionDeleteContext() {
                        this.prepareDeleteContext = false
-                       const res = await this.$store.dispatch('removeContext', { context: this.context, receivers: this.context.sharing })
+                       const context = this.getContext(this.contextId)
+                       const res = await this.$store.dispatch('removeContext', { context, receivers: context.sharing })
                        if (res) {
                                showSuccess(t('tables', 'Application "{context}" removed.', { context: this.title }))
                                // if the active context was deleted, go to startpage
```

File: src/store/store.js [hunk 2]
```diff
@@ -493,11 +494,36 @@ export default new Vuex.Store({
...
 			const contexts = state.contexts
-			const index = contexts.findIndex(t => t.id === id)
+			const index = contexts.findIndex(c => c.id === id)
 			contexts.splice(index, 1)
 			commit('setContexts', [...contexts])
 			return true
 		},
+		async removeContext({ dispatch, state, commit }, { context }) {
+			try {
+				await dispatch('unShareContext', { receivers: context.sharing })
```
Comment (juliushaertl) [inline src/store/store.js:504]: I would actually prefer to just send a delete request and the backend would not only delete the context but also any shares, pages, app menu entries

@blizzz Any objections? Currently the ContextService (called by ContextController::delete) just drops the context entry it seems

Comment (enjeck) [reply src/store/store.js]: it works, thanks @blizzz

File: src/modules/modals/EditContext.vue (hunk 1 above)
Comment (enjeck) [reply src/modules/modals/EditContext.vue]: good catch, thx

**Comment by enjeck** [comment]
@blizzz How can I fix the Static Analysis errors here?

**Comment by blizzz** [comment]
> @blizzz How can I fix the Static Analysis errors here?

Ah, leftovers from my changes. I added a parameter to ContextService, but changed name and type during working on it. Alas not everywhere. So it's fixing, fix-upping the original commit, and rebasing with squashing it.

**Comment by blizzz** [comment]
Done and rebased.

File: src/modules/modals/DeleteContext.vue [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+		},
+		showModal: {
+			type: Boolean,
+			default: false,
+		},
+	},
+	computed: {
+		...mapState(['activeContextId']),
+		getTranslatedDescription() {
+			// TODO confirm that we actually do delete all the shares
```
Comment (blizzz) [inline src/modules/modals/DeleteContext.vue:38]: This TODO can go now, as it is handled in the backend.

**Review by blizzz — APPROVED** [review]

**Review by juliushaertl — APPROVED** [review]

**Pull request merged**